src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1757,7 +1757,7 @@ private function dumpValue($value, bool $interpolate = true): string
 
                     $code = sprintf('return %s;', $code);
 
-                    return sprintf("function ()%s {\n            %s\n        }", $returnedType, $code);
+                    return sprintf("function () use (\$container) %s {\n            %s\n        }", $returnedType, $code);
```

Comment:
This probably should be migrated to a dedicated PR + tests

The previous code trigger a `Notice: Undefined variable: container` exception  (#39851 should also fix this issue).
Here is the dumped code:
```
        return $container->privates['.console.command.cache_clear.lazy'] = new \Symfony\Component\Console\Command\LazyCommand(function () : \Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand {
            return ($container->privates['console.command.cache_clear'] ?? $container->load('getConsole_Command_CacheClearService'));
        });
```

Reply:
Depending on the way the container is dumped, the container should be referenced as `$container` or `$this`.

Look for `use ($container)` in the current `PhpDumper` to figure out how this should be done.

Also for sure, this is a bug fix and it should be submitted on a dedicated PR.

Reply:
see #39873